### ISSUE TITLE
More robust detection of whether or not to run `setup` during server start (even for upgrades)

### ISSUE THREAD

**Issue by jamalex** [learningequality/ka-lite#4069]
For context, see: https://github.com/learningequality/installers/issues/86#issuecomment-119763059

Currently, we auto-run `setup` within the `initialize_kalite` command, before starting the server, when the following conditions are met:

```python
    # Ensure that the database has been synced and a Device has been created
    try:
        assert Settings.get("private_key") and Device.objects.count()
    except (DatabaseError, AssertionError):
        # Otherwise, run the setup command
        self.stdout.write("Setting up KA Lite; this may take a few minutes; please wait!\n")
        call_command("setup", interactive=False)
```

However, this doesn't get triggered during an upgrade, but it needs to be, to avoid problems and confusion. One way we could detect whether or not to auto-run setup would be:
- Have a value called `initialized_version` set in the `Settings` model.
- Check this value in `initialize_kalite`, and run `setup` if it isn't defined or if it's older than the current version.
- Upon completion of the `setup` command, update the value of `initialized_version` if needed.

This would be pretty slick and efficient, unless I'm overlooking something critical. Any thoughts?

**Comment by MCGallaspy** [comment]
Seems reasonable. I'm curious why the Settings model lives in fle_utils instead of the app itself?

**Comment by benjaoming** [comment]
:+1: for this proposal, especially because it removes responsibility from installers.

**Comment by benjaoming** [comment]
tldr; it would be cool to unify these "initialization" logics into one place.. it's about preloading the caches as well... if you feel like moving around code to put everything into one "easy to read" location, that would be really cool... it's used by "kalite start" and "kaserve". The latter should be removed IMO, but "kalite manage runserver" should ideally also be running the same initialization as kalite start... which it doesn't currently. DEBUG should be the only thing that changes how things are initialized / preloaded.

**Comment by jamalex** [comment]
> Seems reasonable. I'm curious why the Settings model lives in fle_utils instead of the app itself?

Old Ben saw it as general purpose enough to move it out there, it seems.
